src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Unsigned.Default.cs

```diff
@@ -2,33 +2,158 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
 namespace System.Buffers.Text
 {
     /// <summary>
     /// Methods to format common data types as Utf8 strings.
     /// </summary>
     public static partial class Utf8Formatter
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool TryFormatUInt64Default(ulong value, Span<byte> buffer, out int bytesWritten)
         {
             if (value < 10)
             {
-                if (buffer.Length == 0) goto FalseExit;
-                buffer[0] = (byte)('0' + value);
-                bytesWritten = 1;
-                return true;
+                return TryFormatUInt32SingleDigit((uint)value, buffer, out bytesWritten);
+            }
+
+            if (IntPtr.Size == 8)    // x64
+            {
+                return TryFormatUInt64MultipleDigits(value, buffer, out bytesWritten);
+            }
+            else    // x86
+            {
+                if (value <= uint.MaxValue)
+                {
+                    return TryFormatUInt32MultipleDigits((uint)value, buffer, out bytesWritten);
+                }
+                else
+                {
+                    if (value <= Utf8Constants.BillionMaxUIntValue)
+                    {
+                        return TryFormatUInt64LessThanBillionMaxUInt(value, buffer, out bytesWritten);
+                    }
+                    else
+                    {
+                        return TryFormatUInt64MoreThanBillionMaxUInt(value, buffer, out bytesWritten);
+                    }
+                }
+            }
```

Comment:
Nit: You could eliminate a lot of vertical space with code like:
```C#
return
    IntPtr.Size == 8 ? TryFormatUInt64MultipleDigits(value, buffer, out bytesWritten) : // x64
    value <= uint.MaxValue ? TryFormatUInt32MultipleDigits((uint)value, buffer, out bytesWritten) :
    value <= Utf8Constants.BillionMaxUIntValue ? TryFormatUInt64LessThanBillionMaxUInt(value, buffer, out bytesWritten) :
    TryFormatUInt64MoreThanBillionMaxUInt(value, buffer, out bytesWritten);
```

Reply:
True. But doing the same to Signed.Default hurts readability quite a bit so I am leaning against it.

```C#
return
    // x64
    IntPtr.Size == 8 ? TryFormatInt64MultipleDigits(value, buffer, out bytesWritten) :
    // x86
    value <= int.MaxValue && value >= int.MinValue ? TryFormatInt32MultipleDigits((int)value, buffer, out bytesWritten) :
    value <= (long)Utf8Constants.BillionMaxUIntValue && value >= -(long)Utf8Constants.BillionMaxUIntValue ?
    value < 0 ? TryFormatInt64MoreThanNegativeBillionMaxUInt(-value, buffer, out bytesWritten) :
    TryFormatUInt64LessThanBillionMaxUInt((ulong)value, buffer, out bytesWritten) :
    value < 0 ? TryFormatInt64LessThanNegativeBillionMaxUInt(-value, buffer, out bytesWritten) :
    TryFormatUInt64MoreThanBillionMaxUInt((ulong)value, buffer, out bytesWritten);
```

I think there is value in keeping the shape between Signed/Unsigned similar. Thoughts?

Reply:
I don't feel strongly about it; whatever you prefer is fine.  FWIW, I find all the nested if/elses to hurt readability as well.